src/modules/oAuth/oAuth.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@nestjs/common';
+import { UsersRepository } from '../users/users.repository';
+import { generateToken } from 'src/shared';
+
+@Injectable()
+export class OAuthService {
+  constructor(readonly userRepository: UsersRepository) {}
+  async googleLogin(req) {
+    const isUserExist = await this.userRepository.getUserBy({
+      email: req.user.email,
+    });
+
+    if (isUserExist) {
+      const accessToken = generateToken(isUserExist.id, '72h');
```

Comment:
you can use jwt service from '@nestjs/jwt' . It already has methods for generating tokens.

Reply:
replaced